src/controller_examples/zookeeper_controller/spec/zookeeper_lib.rs

```diff
@@ -20,4 +20,36 @@ pub struct ZKNodeResultView {
     pub res: Result<(), Error>,
 }
 
+pub struct ZooKeeperState {
+    pub data: Map<StringView, StringView>,
+}
+
+pub open spec fn external_process(input: ZKSupportInputView, state: Option<ZooKeeperState>) -> (Option<ZKSupportOutputView>, Option<ZooKeeperState>) {
+    match input {
+        ZKSupportInputView::ReconcileZKNode(path,uri,replicas) => reconcile_zk_node(path, uri, replicas, state),
+    }
+}
+
+pub open spec fn reconcile_zk_node(
```

Comment:
The current reconcile_zk_node looks fine to me. And I think we should think hard about two problems:
(1) what is our assumption on the external APIs? Do we assume they always return the result successfully, or they might return some transient error (and eventually get back to normal)? Or some other assumptions?
(2) The external API's state and result might depend on the state of the k8s objects. Like, if the zk pods are not ready then the zk API won't work. Should we try to specify such dependencies?

Reply:
(1) This has to do with the specific reconciler. For the zookeeper one, whether to return error result or not is fine. 
(2) Maybe this can be considered when we start to strip the library thing off the controller?